pkg/syncer/specsyncer.go

```diff
@@ -149,20 +149,13 @@ func (c *Controller) applyToDownstream(ctx context.Context, gvr schema.GroupVers
 	obj.SetResourceVersion("")
 	obj.SetNamespace(namespace)
 	obj.SetManagedFields(nil)
+	// Strip owner references, to avoid orphaning by broken references,
+	// and make sure cascading deletion is only performed once upstream.
+	obj.SetOwnerReferences([]metav1.OwnerReference{})
+	// Strip finalizers to avoid the deletion of the downstream resource from being blocked.
+	obj.SetFinalizers([]string{})
```

Comment:
there will be finalizers added by downstream controllers. What happens with them?

Reply:
As Server-Side Apply is used to update the object downstream, and the `finalizers` key is removed from the SSA patch, the downstream finalizers are left untouched, so that controllers responsible for these finalizers can perform their logic.

I've added a commit that sets the deletion propagation policy to `Foreground`, so that the deletion cascades the entire ownership graph, and the `deletionTimestamp` field is set, so that the object remains in the key-value store until all its finalizers are removed.